global/scripts/golang/test/run.sh

```diff
@@ -12,12 +12,8 @@ export INIT_SCRIPT="config.sh"
 
 touch coverage.xml
 
-# determine the directory to use
-dir="cmd"
-[ -d "$(pwd)/main" ] && dir="main"
-
 # Run the tests
-go test -v -tags test,unit,integration -coverpkg ./$dir/... -covermode=count ./$dir/... -coverprofile=coverage.txt
+go test -v -tags test,unit,integration -coverpkg ./... -covermode=count ./... -coverprofile=coverage.txt
```

Comment:
Well, I was trying to check because this way you are going to run in the tests folder as well, where you have the stubs. Instead, you should try to understand what is the root folder, before running.

Reply:
Check here the workaround I did: https://github.com/rios0rios0/pipelines/blob/main/gitlab/golang/stages/35-management/go.yaml#L22

Reply:
That's not really necessary, and probably not ideal either. For Go there's `cmd` and `internal` and other folders that can contain code, and you're using `main` so that's another folder. Instead of that we already have `sonar.coverage.exclusions=**/*_test.go,test/**` in `sonar-project.properties` which excludes the test files. We can also optionally add more to exclude anything else that we don't need. `./...` looks fine to me.